code/modules/xenoarcheaology/tools/tools.dm

```diff
@@ -310,3 +310,170 @@
 		usr << browse(null, "window=locater")
 
 	updateSelfDialog()
+
+/obj/item/device/xenoarch_multi_tool
+	name = "xenoarcheology multitool"
+	desc = "Has the features of the Alden-Saraspova counter, a measuring tape, and a depth analysis scanner all in one!"
+	icon_state = "ano_scanner2"
+	item_state = "lampgreen"
+	icon = 'icons/obj/xenoarchaeology.dmi'
+	origin_tech = list(TECH_MAGNET = 3, TECH_ENGINEERING = 3, TECH_BLUESPACE = 2)
+	matter = list(DEFAULT_WALL_MATERIAL = 10000,"glass" = 5000)
+	w_class = ITEMSIZE_SMALL
+	slot_flags = SLOT_BELT
+	//Multitool specific
+	var/scanning = 1 //Lets start on scanning mode.
+	var/measuring = 0
+
+	//Counter Specific
+	var/last_scan_time = 0
+	var/scan_delay = 25
+
+	//Depth Specific
+	var/list/positive_locations = list()
+	var/datum/depth_scan/current
+
+/obj/item/device/xenoarch_multi_tool/attack_self(var/mob/living/user)
+	var/choice
+	if(!scanning) //Are they not in the view records mode?
+		choice = input(user, "What do you wish to do?") as null|anything in list("Scan/View Records", "Measure", "Detect")
+
+	if(choice == "Scan/View Records" || scanning)
+		if(!scanning)
+			to_chat(usr,"You switch the multitool to scanning mode. Use again to view scan records.")
+			scanning = 1
+			measuring = 0 //Just to make sure!
+			return
+		else
```

Comment:
Instead of building the whole datum in exactly the same manner as the depth scanner, you could just tell an internal DS to build it for you